File: output/CodeLlama/avatar/Python/Java/codeforces_106_B.java
import java.util.*;
public class codeforces_106_B {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        int n = sc.nextInt();
        int[][] input_array = new int[n][4];
        for (int i = 0; i < n; i++) {
            input_array[i][0] = sc.nextInt();
            input_array[i][1] = sc.nextInt();
            input_array[i][2] = sc.nextInt();
            input_array[i][3] = sc.nextInt();
        }
        int[][] outdated = new int[n][4];
        for (int i = 0; i < n; i++) {
            for (int j = 0; j < n; j++) {
                if (input_array[i][0] < input_array[j][0] && input_array[i][1] < input_array[j][1] && input_array[i][2] < input_array[j][2]) {
                    outdated[i][0] = input_array[i][0];
                    outdated[i][1] = input_array[i][1];
                    outdated[i][2] = input_array[i][2];
                    outdated[i][3] = input_array[i][3];
                }
            }
        }
        int minimum_cost = 100000;
        int input_number = 0;
        for (int i = 0; i < n; i++) {
            if (input_array[i][3] < minimum_cost) {
                minimum_cost = input_array[i][3];
                input_number = i + 1;
            }
        }
        System.out.println(input_number);
    }
}

File: output/CodeGen/avatar/Java/Python/codeforces_106_B.py
import java.util.Arrays ; import java.util.* ; import java.io.PrintWriter ; import java.util.Scanner ; 
def main ( ): 
    input = java.io.BufferedReader ( java.io.InputStreamReader ( System.in ) ) ; 
    output = java.io.PrintWriter ( java.io.OutputStreamWriter ( System.out ) ) ; 
    n = int ( input.readLine ( ) ) ; 
    vec = [ [ int ( x ) for x in input.readLine ( ).split ( " " ) ] for i in range ( n ) ] ; 
    ans = 500 ; 
    ansprice = 20000 ; 
    for i in range ( n ): 
        fl = 1 ; 
        for j in range ( n ): 
            if ( vec [ i ] [ 0 ] < vec [ j ] [ 0 ] and vec [ i ] [ 1 ] < vec [ j ] [ 1 ] and vec [ i ] [ 2 ] < vec [ j ] [ 2 ] ): 
                fl = 0 ; 
        if ( fl == 1 ): 
            if ( vec [ i ] [ 3 ] < ansprice ): 
                ansprice = vec [ i ] [ 3 ] ; 
                ans = i + 1 ; 
    output.println ( ans ) ; 
    output.close ( ) ; 


import java.util.Arrays ; import java.util.* ; import java.io.PrintWriter ; import java.util.Scanner ; 
public class codeforces_106_B { 
    public static void main ( String [ ] args ) { 
        Scanner input = new Scanner ( System.in ) ; 
        PrintWriter output = new PrintWriter ( System.out ) ; 
        int n = input.nextInt ( ) ; 
        int [ ] [ ] vec = new int [ n ] [ 4 ] ; 
        for ( int i = 0 ; i < n ; i ++ ) { 
            vec [ i ] [ 0 ] = input.nextInt ( ) ; 
            vec [ i ] [ 1 ] = input.nextInt ( ) ; 
            vec [ i ] [ 2 ] = input.nextInt ( ) ; 
            vec [ i ] [ 3 ] = input.nextInt ( ) ; 
        } 
        int ans = 500 ; 
        int ansprice = 20000 ; 
        for ( int i = 0 ; i < n ; i ++ ) { 
            int fl = 1 ; 
            for ( int j = 0 ; j < n ; j ++ ) { 
                if ( vec [ i ] [ 0 ] < vec [ j ] [ 0 ] and vec [ i ] [ 1 ] < vec [ j ] [ 1 ] and vec [ i ] [ 2 ] < vec [ j ] [ 2 ] ) { 
                    fl = 0 ; 
                } 
            } 
            if ( fl == 1 ) { 
                if ( vec [ i ] [ 3 ] < ansprice ) { 
                    ansprice = vec [ i ] [ 3 ] ; 
                    ans = i + 1 ; 
                } 
            } 
        } 
        output.println ( ans ) ; 
        output.close ( ) ; 
    } 
}